File: python/ccpayment/services/merchant_assets.py
"""Merchant Assets Service"""


class MerchantAssetsService:
    def __init__(self, client):
        self.client = client

    def get_app_coin_asset_list(self):
        return self.client._post("/getAppCoinAssetList", {})

    def get_app_coin_asset(self, coin_id: int):
        return self.client._post("/getAppCoinAsset", {"coinId": coin_id})

    def get_app_collect_fee_record_list(self, data=None):
        return self.client._post("/getAggregationFeeRecord", data or {})

File: javascript/src/services/merchant-assets.js
class MerchantAssetsService {
  constructor(client) { this.client = client; }
  async getAppCoinAssetList() { return this.client.post('/getAppCoinAssetList', {}); }
  async getAppCoinAsset(coinId) { return this.client.post('/getAppCoinAsset', { coinId }); }
  async getAppCollectFeeRecordList(data = {}) { return this.client.post('/getAggregationFeeRecord', data); }
}
module.exports = MerchantAssetsService;
